src/Editor/Resources.cpp

```diff
@@ -1,18 +1,36 @@
 #include "Resources.hpp"
 
-#include <json/value.h>
 #include <Engine/Texture/TextureAsset.hpp>
 #include <Engine/Geometry/Model.hpp>
 #include <Engine/Audio/SoundBuffer.hpp>
+#include <Engine/Script/ScriptFile.hpp>
 #include <Engine/Hymn.hpp>
 #include <Engine/Manager/Managers.hpp>
 #include <Engine/Manager/ResourceManager.hpp>
 #include <Engine/Util/FileSystem.hpp>
 
 using namespace std;
 
+string ResourceList::Resource::GetName() const {
+    switch (type) {
+    case Type::SCENE:
+        return scene;
+    case Type::MODEL:
+        return model->name;
+    case Type::TEXTURE:
+        return texture->name;
+    case Type::SOUND:
+        return sound->name;
+    case Type::SCRIPT:
+        return script->name;
+    default:
+        return "";
+    }
+}
+
 ResourceList::ResourceList() {
-    activeScene = -1;
+    activeScene = "";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'activeScene' is assigned in constructor body. Consider performing initialization in initialization list. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuseInitializationList)